graylog2-web-interface/src/logic/permissions/EntityShareState.tsx

```diff
@@ -61,36 +62,36 @@ const _sortAndOrderGrantees = <T: GranteeInterface>(grantees: Immutable.List<T>,
     .filter((grantee) => activeShares && activeShares.findIndex((activeShare) => activeShare.grantee === grantee.id) === -1)
     .reverse();
 
-  return Immutable.List().concat(
+  return Immutable.List<T>().concat(
     newGrantees,
     granteesByType.get('error'),
     granteesByType.get('global'),
     granteesByType.get('team'),
     granteesByType.get('user'),
-  ).filter((grantee) => grantee);
+  ).filter((grantee) => !!grantee).toList();
 };
 
-type InternalState = {|
+type InternalState = {
   entity: GRN,
   availableGrantees: GranteesList,
   availableCapabilities: CapabilitiesList,
   activeShares: ActiveShares,
   selectedGranteeCapabilities: SelectedGranteeCapabilities,
   missingDependencies: MissingDependencies,
   validationResults: ValidationResult,
-|};
+};
 
-export type EntityShareStateJson = {|
+export type EntityShareStateJson = {
```

Comment:
There are a bunch of `ESLint: Identifier 'available_grantees' is not in camel case.(camelcase)` in here, in case you want to take care of them :)